server/src/main/java/org/opensearch/action/support/TransportIndicesResolvingAction.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.support;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.cluster.metadata.ResolvedIndices;
+
+/**
+ * An additional interface that should be implemented by TransportAction implementations which need to resolve
+ * IndicesRequests or other action requests which specify indices. This interface allows other components to retrieve
+ * precise information about the indices an action is going to operate on. This is particularly useful for access
+ * control implementations, but can be also used for other purposes, such as monitoring, audit logging, etc.
+ * <p>
+ * Classes implementing this interface should make sure that the reported indices are also actually the indices
+ * the action will operate on. The best way to achieve this, is to move the index extraction code from the execute
+ * methods into reusable methods and to depend on these both for execution and reporting.
+ */
+public interface TransportIndicesResolvingAction<Request extends ActionRequest> {
```

Comment:
Using an interface is a good approach for exposing index resolution logic, rather than requiring each plugin to implement it independently. I like the idea of having the core provide this functionality.

@nibix @cwperks The interface can also be extended with other helpful methods for plugins, such as:
`getOperationType`, `requiresClusterState` or a method to see if the action involves system indices.

Additionally, we could consider renaming the interface to `TransportIndicesAwareAction`, making it more flexible and future-proof for supporting additional capabilities.

Reply:
Or we could start with `TransportIndicesAwareAction` interface first and then if there's a need for more retrieval methods in the future, we can decide on if creating a new interface or adding in `TransportIndicesAwareAction` would be better.

Reply:
Sounds good to me! :+1: